### PR TITLE
Syncing fix

### PR DESCRIPTION
**Pull request by arkpar**

### CODE REVIEW

File: util/src/network/host.rs [hunk 1]
```diff
@@ -664,7 +664,6 @@ impl<Message> Host<Message> where Message: Send + Sync + Clone {
 					Err(e) => {
 						trace!(target: "network", "Session read error: {}:{:?} ({:?}) {:?}", token, s.id(), s.remote_addr(), e);
 						match e {
-							UtilError::Network(NetworkError::Disconnect(DisconnectReason::UselessPeer)) |
```
Comment (arkpar) [inline util/src/network/host.rs]: Geth sends UselessPeer on ping timeout so we don't want to ban for that

File: sync/src/chain.rs [hunk 2]
```diff
@@ -1128,14 +1131,19 @@ impl ChainSync {
...
+		let mut aborting = Vec::new();
 		for (peer_id, peer) in &self.peers {
 			if peer.asking != PeerAsking::Nothing && (tick - peer.ask_time) > CONNECTION_TIMEOUT_SEC {
 				trace!(target:"sync", "Timeout {}", peer_id);
 				io.disconnect_peer(*peer_id);
+				aborting.push(*peer_id);
 			}
 		}
+		for p in aborting {
+			self.on_peer_aborting(io, p);
```
Comment (arkpar) [inline sync/src/chain.rs]: Abort peers immediately, don't wait for disconnect event from network

File: sync/src/chain.rs [hunk 3]
```diff
@@ -737,6 +738,8 @@ impl ChainSync {
 				None => {
 					// TODO: get hash by number from the block queue
 					trace!(target: "sync", "Could not revert to previous block, last: {} ({})", self.last_imported_block, self.last_imported_hash);
+					// just wait for full sync to complete
+					self.pause_sync();
```
Comment (gavofyork) [inline sync/src/chain.rs]: looks dangerous. what "unpauses" the sync?

Comment (arkpar) [inline sync/src/chain.rs]: periodic check in `maintain_sync`

Comment (gavofyork) [inline sync/src/chain.rs]: *buries head in sand and hopes for the best*

**Pull request merged**